idem_provider_azurerm/states/azurerm/resource/management_lock.py

```diff
@@ -632,6 +658,8 @@
         return ret
 
     ret['comment'] = 'Failed to create management lock {0}! ({1})'.format(name, lock.get('error'))
+    if ret['result'] == False:
```

Comment:
`if not variable` unless None is a possible result and it's different from False